### DISCUSSION TITLE
yarn rebuild-native -- --force-rebuild=true 报错

### DISCUSSION THREAD

**Discussion by caronchen** [opensumi/core#2273] [Q&A]
执行 `yarn rebuild-native -- --force-rebuild=true` 时始终报错，不知道是什么原因？

错误如下：

`$ node ./scripts/rebuild-native.js --target=electron --force-rebuild=true
rebuilding native for electron version 18.3.4
rebuilding node-pty
npm_config_arch=x64 npm_config_target_arch=x64 node-gyp rebuild --openssl_fips=X --target=18.3.4 --arch=x64 --dist-url=https://electronjs.org/headers
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.1
gyp info using node@19.6.0 | darwin | x64
gyp info find Python using Python version 3.9.7 found at "/usr/local/opt/python@3.9/bin/python3.9"
gyp http GET https://electronjs.org/headers/v18.3.4/node-v18.3.4-headers.tar.gz
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: This is most likely not a problem with node-gyp or the package itself and
gyp ERR! stack is related to network connectivity. In most cases you are behind a proxy or have bad
gyp ERR! stack network settings.
gyp ERR! stack     at go (/Users/zf/opensumi/ide-electron/node_modules/node-gyp/lib/install.js:173:17)
gyp ERR! stack     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
gyp ERR! stack     at async install (/Users/zf/opensumi/ide-electron/node_modules/node-gyp/lib/install.js:62:18)
gyp ERR! System Darwin 22.1.0
gyp ERR! command "/usr/local/bin/node" "/Users/zf/opensumi/ide-electron/node_modules/.bin/node-gyp" "rebuild" "--openssl_fips=X" "--target=18.3.4" "--arch=x64" "--dist-url=https://electronjs.org/headers"
gyp ERR! cwd /Users/zf/opensumi/ide-electron/node_modules/node-pty
gyp ERR! node -v v19.6.0
gyp ERR! node-gyp -v v9.3.1
gyp ERR! not ok
node:child_process:960
    throw err;
    ^

Error: Command failed: npm_config_arch=x64 npm_config_target_arch=x64 node-gyp rebuild --openssl_fips=X --target=18.3.4 --arch=x64 --dist-url=https://electronjs.org/headers`

**Comment by yantze** [comment]
可能是网络错误，切换到 ide-electron 仓库的 main-cn 分支试试

**Reply by caronchen** [reply]
> 可能是网络错误，切换到 ide-electron 仓库的 main-cn 分支试试

已经是 main-cn 分支。

<img width="739" alt="image" src="https://user-images.githubusercontent.com/150803/218389432-8946ab32-d39e-42fd-a8aa-768570033dfa.png">

**Comment by yantze** [comment]
```
gyp ERR! stack is related to network connectivity. In most cases you are behind a proxy or have bad
gyp ERR! stack network settings.
```
看起来还是网络问题，试试开个代理，另外一个方式是切换到 `tmp/electron-dist-cn` 分支，试试 electron 头文件地址也切换为镜像地址
